spec/controllers/api/irs_attempts_api_controller_spec.rb

```diff
@@ -107,6 +79,72 @@
         expect(Base64.strict_decode64(response.headers['X-Payload-Key'])).to be_present
         expect(response.body).to eq(test_object)
       end
+
+      context 'with aws_s3_stream enabled' do
+        let(:test_object) { '{test: "1234567890 12345"}' }
+        before do
+          allow(IdentityConfig.store).to receive(:irs_attempt_api_aws_s3_stream_enabled).
+            and_return(true)
+          allow(IdentityConfig.store).to receive(:irs_attempt_api_aws_s3_stream_buffer_size).
+            and_return(10)
+
+          Aws.config[:s3] = {
+            stub_responses: {
+              head_object: { content_length: test_object.bytesize },
+              get_object: proc do |context|
+                range_string = context.params[:range]
+                byte_string = range_string.split('=')[1]
+                byte_range_array = byte_string.split('-')
+                { body: test_object.byteslice(
+                  byte_range_array[0].to_i,
+                  IdentityConfig.store.irs_attempt_api_aws_s3_stream_buffer_size + 1,
+                ) }
+              end,
+            },
+          }
+        end
+
+        it 'should render data streamed from s3 correctly' do
+          post :create, params: { timestamp: time.iso8601 }
+
+          expect(response).to be_ok
+          expect(Base64.strict_decode64(response.headers['X-Payload-IV'])).to be_present
+          expect(Base64.strict_decode64(response.headers['X-Payload-Key'])).to be_present
+          expect(response.content_type).to eq('application/octet-stream')
+          expect(response['Content-Disposition']).
+            to eq("attachment; filename=\"test_filename\"; filename*=UTF-8''test_filename")
+
+          expect(response.stream.body).to eq(test_object)
+        end
+      end
+    end
+
+    context 'with timestamp problems' do
+      it 'returns unprocessable_entity with no timestamp' do
```

Comment:
```suggestion
      it 'returns unprocessable_entity when given no timestamp' do
```

You just moved this test, so this is kind of unfair feedback, but my eyes read "with" as referring to what we return, not what we were given, and I was confused.

Reply:
I agree!